pkg/cluster/enableendpoints.go

```diff
@@ -0,0 +1,44 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/util/subnet"
+)
+
+// enableServiceEndpoints should enable service endpoints on
+// subnets for storage account access
+func (m *manager) enableServiceEndpoints(ctx context.Context) error {
+	subnetIds, err := m.getSubnetIds()
+	if err != nil {
+		return err
+	}
+
+	subnets, err := m.subnet.GetAll(ctx, subnetIds)
+	if err != nil {
+		return err
+	}
+
+	subnetsToBeUpdated := subnet.AddEndpointsToSubnets(api.SubnetsEndpoints, subnets)
+
+	return m.subnet.CreateOrUpdateSubnets(ctx, subnetsToBeUpdated)
```

Comment:
what is the reason behind this not being a single function in the subnet package ? `CreateOrUpdateSubnetsFromIds` would probably be a good name for such a function.

Reply:
The main reason is that if we had just one function like "CreateOrUpdateSubnetsFromIds" it would be a function that 
 both will have business logic + will talk with Azure (infrastructure). 

As is now, subnet.CreateOrUpdateSubnets(...) just talks to Azure (infrastructure) and subnet.AddEndpointsToSubnets(...) just have business logic. This makes subnet.AddEndpointsToSubnets() be easier to test as doesn't have side effects and also separates clear responsibilities between the functions.

Reply:
look, you have just shifted the responsibility elsewhere here, and now you need to export a bunch of functions from `subnet`. `enableServiceEndpoint` has both business logic and talks to azure, so you don't have an argument here. create a new function like this in the subnet package. 

```go 
func createFromIds() error {
	subnets, err := m.subnet.GetAll(ctx, subnetIds)
	if err != nil {
		return err
	}

	subnetsToBeUpdated := subnet.AddEndpointsToSubnets(api.SubnetsEndpoints, subnets)

	return m.subnet.CreateOrUpdateSubnets(ctx, subnetsToBeUpdated)
}
```

and you will be able to mock it as bonus.

Reply:
I thought you meant to move all the logic into the same function instead of creating a function that invokes the already created functions, as I see here. Doing it now.